build/media-suite_compile.sh

```diff
@@ -2432,27 +2438,107 @@ if [[ $mpv != n ]] && pc_exists libavcodec libavformat libswscale libavfilter; t
+                meson_opts+=("${option/--disable-/-D}=disabled")
+                do_simple_print "${orange}Disabling PDF manual, currently broken"'!'"${reset}"
+            elif [[ $option =~ ^--lua=lua* ]]; then
+                # `meson setup build` does not take kindly to a format not matching lua(jit|\d\.\d)
+                local luaver="${option/--lua=lua}"
+                if [[ luaver =~ ^(5|-5) ]]; then
+                    luaver='5.${luaver:-1}'
+                elif [[ luaver =~ ^jit ]]; then
+                    luaver='jit'
+                fi
+                meson_opts+=("-Dlua=lua${luaver}")
+                # Remove the `-Dlua=enabled` flag if present, since the order of flags matters.
+                # - `-Dlua=lua5.2 -Dlua=enabled` makes Meson behave like only `-Dlua=enabled` was specified;
+                # - `-Dlua=enabled -Dlua=lua5.2` makes Meson take lua5.2 if available, erroring otherwise;
+                meson_opts=("${meson_opts[@]/ -Dlua=enabled}")
+            elif [[ $option =~ ^--enable-lua ]] && ! [[ "${meson_opts[@]}" =~ -Dlua= ]]; then
+                meson_opts+=("-Dlua=enabled")
+            fi
+            unset option
+        done
+        do_simple_print "${meson_opts[@]}"
+
         extra_script pre configure
-        CFLAGS+=" ${mpv_cflags[*]} -Wno-int-conversion" LDFLAGS+=" ${mpv_ldflags[*]}" \
+        local old_ldf="$LDFLAGS"
+        CFLAGS+=" ${mpv_cflags[*]} -Wno-int-conversion" LDFLAGS+=" ${mpv_ldflags[*]} -larchive -lb2 -lzstd" \
             RST2MAN="${MINGW_PREFIX}/bin/rst2man" \
             RST2HTML="${MINGW_PREFIX}/bin/rst2html" \
-            RST2PDF="${MINGW_PREFIX}/bin/rst2pdf2" \
-            PKG_CONFIG="$LOCALDESTDIR/bin/ab-pkg-config" \
-            log configure /usr/bin/python waf configure \
-            "--prefix=$LOCALDESTDIR" "--bindir=$LOCALDESTDIR/bin-video" \
-            "${MPV_OPTS[@]}"
+            RST2PDF="${MINGW_PREFIX}/bin/rst2pdf2"
+        mkdir build
+        PKG_CONFIG="pkgconf --keep-system-libs --keep-system-cflags" CC=${CC/ccache /}.bat CXX=${CXX/ccache /}.bat \
+            log "meson.setup" meson setup build "${meson_opts[@]}" --prefix="${LOCALDESTDIR}" --bindir="${LOCALDESTDIR}/bin-video" --default-library="${default_lib}" --buildtype=release
         extra_script post configure
 
-        replace="LIBPATH_lib\1 = ['${LOCALDESTDIR}/lib','${MINGW_PREFIX}/lib']"
-        sed -r -i "s:LIBPATH_lib(ass|av(|device|filter)) = .*:$replace:g" ./build/c4che/_cache.py	
-
         extra_script pre build
-        log build /usr/bin/python waf -j "${cpuCount:-1}"
+        log "meson.compile" meson compile -C build
         extra_script post build
 
         extra_script pre install
-        log install /usr/bin/python waf -j1 install ||
-            log install /usr/bin/python waf -j1 install
+        cpuCount=1 log "meson.install" meson install -C build &&
+        mv "${LOCALDESTDIR}/bin-video/libmpv-2.dll" "$LOCALDESTDIR/bin-video/mpv-2.dll"
```

Comment:
I can't find a way to tell Meson I don't want the output file to be named `libmpv-2.dll`, hence the manual move